tests/functional_tests.rs

```diff
@@ -98,12 +100,13 @@ async fn main() {
         },
     ];
 
-    Provision::new("my-hostname".to_string(), User::new(username, keys))
-        .hostname_provisioners([HostnameProvisioner::Hostnamectl])
-        .user_provisioners([UserProvisioner::Useradd])
-        .password_provisioners([PasswordProvisioner::Passwd])
-        .provision()
-        .expect("Failed to provision host");
+    Provision::new(
```

Comment:
So the builder API is completely deprecated now that we have a configuration file?

Reply:
I wouldn't say it's entirely deprecated, but you're right in that the builder API is not essential now for things like hostname, username, and password settings. We could still find a way to use it for runtime-specific overrides or maybe testing scenarios if we wanted. 

I will also note that losing some of the builder API’s flexibility was the biggest drawback of the design change to add a Config struct into Provision, but I chose this approach because it keeps things consistent and allows us to easily override values in the config when adjustments are needed. It also centralizes everything, so we don’t end up with two separate Config structs.